models/Result.js

```diff
@@ -280,78 +133,38 @@ class Result {
         });
         const filteredResultsIds = filteredResults.sort(resultsSort).map(({ id: resultId }) => resultId);
 
-        const switchQuery = client.query(switchSql, [filteredResultsIds]);
-
-        switchQuery.on('end', () => {
-          const oneUserQuery = client.query(oneUserSql, [onePersonResultIds]);
-
-          oneUserQuery.on('end', () => {
-            client.end();
-            resolve();
-          });
-          oneUserQuery.on('error', err => {
-            client.end();
-            logger.errorLog('Results.switchSpotsForPerformance: oneUserQuery', err);
-            reject(err);
-          });
-        });
-
-        switchQuery.on('error', err => {
-          client.end();
-          logger.errorLog('Results.switchSpotsForPerformance: switchQuery', err);
-          reject(err);
-        });
-      });
-
-      resultsQuery.on('error', err => {
-        logger.errorLog('Results.switchSpotsForPerformance', err);
-        client.end();
-        reject(err);
-      });
+        db.query(switchSql, [filteredResultsIds])
+        .then(() => {
+          db.query(oneUserSql, [onePersonResultIds])
+          .then(resolve)
+          .catch(reject);
+        })
+        .catch(reject);
+      })
+      .catch(reject);
     });
   }
 
   static update(attributes) {
-    const client = db.createClient();
-
-    return new Promise((resolve, reject) => {
-      client.connect();
-      client.on('error', (err) => reject(err));
-
-      const id = attributes.id;
+    const id = attributes.id;
 
-      if (typeof id === 'undefined') {
-        reject(new Error('No id provided with attributes'));
-      }
+    if (typeof id === 'undefined') {
+      return Promise.reject(new Error('No id provided with attributes'));
+    }
 
-      delete attributes.id;
-      const { sql, values } = db.queryBuilder(Result, attributes, { statement: 'UPDATE', id });
-      const query = client.query(sql, values);
+    delete attributes.id;
```

Comment:
You're mutating a passed argument here. I don't know where it comes from, could this get you in trouble elsewhere? Might be safer to copy needed props out to a new object. Even if it's just a lodash.omit(['id']); or even const copied = Object.assign({}, attributes); delete copied.id;